### PR TITLE
EVG-20527 introduce PutMany

### PR DESCRIPTION
**Pull request by ybrill**
[EVG-20527](https://jira.mongodb.org/browse/EVG-20527)

### Description
https://github.com/mongodb/amboy/pull/330 added a PutMany to amboy. This PR updates amboy to get it and uses it to great effect in `cronsRemoteFifteenSecond`. If this looks good I'll do the same for the other populators.

### Testing
I'll take it for a spin on staging when I can get the lock.

### CODE REVIEW

**Comment by ablack12** [comment]
Smallest nit: can you call this PR title something like "introduce PutMany" to stick with PR conventions?

File: units/crons.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ const (
 	podCreationQueueGroup           = "service.pod.create"
 )
 
+type jobFactory func(context.Context, time.Time) ([]amboy.Job, error)
```
Comment (julianedwards) [inline units/crons.go:40]: Nit: maybe call this `cronJobFactory`?

**Review by julianedwards — APPROVED** [review]
LGTM % nit!

File: units/crons.go (hunk 1 above)
Comment (ybrill) [reply units/crons.go]: done

**Comment by ybrill** [comment]
On staging [scheduler jobs are continuing to run](https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen/result/9bGpNv928Y7).

**Pull request merged**